[Issue 221] Open source group kick deliverable specification draft (#980)
HHS/simpler-grants-gov: documentation/deliverables/individual_deliverables/open-source-group-kickoff.md

> # Open Source Group Kickoff

Reviewer: ```suggestion
# Open Source Onboarding Process
```
Since we're mainly focusing on implementing the comms tools for open source group and setting up the process to onboard new open source participants, I wonder if this is a more accurate title?

| Field           | Value                                                        |
| --------------- | ------------------------------------------------------------ |
| Document Status | Draft                                                        |
| Epic Link       | [Issue 72](https://github.com/HHS/grants-equity/issues/72)   |

> | Epic Dashboard  | [Milestone Roadmap](https://github.com/orgs/HHS/projects/12) |

Reviewer: Can you update this to match the [summary table in the other deliverable specs](https://github.com/HHS/simpler-grants-gov/blob/main/documentation/deliverables/individual_deliverables/static_site_public_launch_infrastructure_content.md#static-site-public-launch-infrastructure-and-content)?

Author: done! Billy, I wasn't sure if I should mark you as a product owner or as another document owner since you would be contributing to this document too. Feel free to change it.

Reviewer: This isn't as relevant if we'll be moving these into the public wiki!

## Short description

> -**What:** Set up the tools and processes needed to build an open source community around the Simpler Grants.gov initiative

Reviewer 2: Nit: for some reason the rendered formatting here for the bullets isn't working. Maybe need a space between `-` and `**`

-**Why:** Ensures that open source contributors and the general public can easily participate in the project and provide input and code
-**Who:** Any public members who want to contribute to the open source project. Focus is on: 

> - Open source contributors

Reviewer 2: Need more indents here?

- Other external stakeholders

## Goals

### Business description & value

**Problem statement:** Currently, the Simpler.Grants.gov project lacks a ready-to-use onboarding experience for individuals interested in joining the open source community. The absence of essential tools, processes, onboarding steps, and guidelines poses a challenge to meeting business goals and mission alignment. 

**Value:** We seek to build stakeholder trust, promote project transparency, and establish a dynamic, participant-driven advisory council Launch an open source community with a primary emphasis on critical components that lay the groundwork for future expansion and productive collaboration, ensuring streamlined community management, fostering a vibrant open source community, and promoting project transparency.

This effort allows us to... 

- Set up the foundational set of communication channels for continuous user input, feedback, and engagement for an open source project
- Ensure that public code contributions meet code quality and security standards
- Build stakeholder trust in our product roadmap and our approach to development

> - Establish an ongoing Participant Advisory Council to guide future development

Reviewer 2: Is this bullet more appropriate for the PAC Kickoff 30k?

Author: I can be more clear here. I was thinking if we wanted to have an open source group within the PAC, but maybe that's not needed or a priority for this.

Reviewer: Agree that this seems a bit outside the scope of this 30k as it's currently defined, but a good goal to save for [that 30k](https://github.com/HHS/simpler-grants-gov/issues/327)

> - Set up the foundational set of communication channels for continuous user input, feedback, and engagement for an open source project
> - Ensure that public code contributions meet code quality and security standards
> - Build stakeholder trust in our product roadmap and our approach to development
> - Establish an ongoing Participant Advisory Council to guide future development

Reviewer: ```suggestion
- Set up the foundational set of communication channels for continuous user input, feedback, and engagement for an open source project
- Ensure that public code contributions meet code quality and security standards
- Build stakeholder trust in our product roadmap and our approach to development
- Streamline and standardize the onboarding process for new open source contributors
```
Could be good to highlight that an important goal of this 30k is to standardize and increase the efficiency of our onboarding process

### User stories

- As a **full-time HHS staff member**, I want to:

>   -  have assurance that the open source channels and methods are configured in a way that eliminates security risks.

Reviewer: Love this! Good callout

- As a **member of an HHS contractor team**, I want to:

>   -  have streamlined tools and processes for managing contributions, reviewing code, and ensuring code quality, so I can efficiently maintain and enhance the project while upholding its standards.

Reviewer: ```suggestion
  -  have streamlined tools and processes for managing contributions, reviewing code, and ensuring code quality, so I can efficiently maintain and enhance the project while upholding its standards.
  - to have a standardized process for onboarding new open source contributors, so that members of the public can be added to our communication channels smoothly and efficiently.
```

>   - to have the ability to monitor the health and inclusivity of the project by tracking metrics like community engagement, issue resolution times, and the diversity of contributors, so I can make informed decisions for community growth and sustainability.

Reviewer: Yes! Love this too!

### Team meetings and open source events

Detail the aspects of scheduling and conducting team scrum meetings. This can include information about the chosen video conferencing tools, meeting frequency, methods for the general public to join open meetings, and any requirements for ensuring effective remote communication during these meetings. 

### Getting Started Guide Development  

There is clear documentation and guidelines for the general public to get started and contribute. Documentation typically should include project overviews, installation instructions, usage guidelines, contribution procedures, and community engagement details, fostering a collaborative and informed open source community. 

### Developer Tools Setup  

Provide technical instructions for setting up developer tools and environments. This can encompass version control systems, code review tools, and any collaborative platforms for coding and testing.

> 
> 

Reviewer: ```suggestion
```
Just removing some of the extra spaces

### Definition of done

- [ ] Onboarding experience ready for the public this includes:
    - [ ] onboarding to the selected chat tool, Slack
    - [ ] onboarding to Github
    - [ ] materials such as a deck or another format for getting started with the project
    - [ ] onboarding guidance for developer tools and environments
- [ ] The selected video conference tool, Zoom, is procured and ready for use for webinars, office hours, or other open source events. 

### Proposed metrics for measuring goals/value/definition of done

- [ ] Number of users onboarded to the open source community

> - [ ] Time to onboard to the open source community

Author: I will include this in the security section but I believe we will need to get security approval to use Google Groups and maybe Zoom since they will be collecting email addresses of users. Slack and wiki are used currently so I assume they are OK but we may need to check on that too.

> - [ ] Time to onboard to the open source community

Author: Added some of your comments into the spec!

## Planning

### Assumptions & dependencies
<!-- Required -->

What capabilities / milestones do we expect to be in place at the beginning of work
on this milestone?

- Procurement of the tools should be in place as the procurement process can delay delivery. Zoom, Slack, and Github are already licensed on the project. 

Are there any notable capabilities / milestones we do NOT expect to be in place at the
beginning of work on this milestone?

> 

Reviewer: ```suggestion
- **Participant advisory council (PAC):** We will not have a participant advisory council in place when work begins on this deliverable. Instead, this deliverable will be required to onboard members of the PAC once we begin work on setting it up.
- **Site content translations:** We will not yet have a process in place to translate the content of our static site or repository documents. That will be addressed in a future 30k ft deliverable.
```

### Open questions
<!-- Optional -->

> - [ ] Is it important to have a public wiki available and ready in this deliverable?

Reviewer: I would say yes, though we can have a very minimal first version of the public wiki.

- [ ] Is it important to have a public API documentation ready for consumption in this deliverable? 

### Not doing
<!-- Optional -->

The following work will *not* be completed as part of this milestone:

> - tools that are not stated as required (Google Groups, public wiki, etc) 

Author: I will include this in the security section but I believe we will need to get security approval to use Google Groups and maybe Zoom since they will be collecting email addresses of users. Slack and wiki are used currently so I assume they are OK but we may need to check on that with the Security team too

## Integrations

### Translations
<!-- Required -->

> 

Reviewer: @lucasmbrown-usds Do we anticipate creating translations of the content in the public wiki?

I would say actually translating the content is out of scope for this 30k, but we should make a note of it if we expect the public wiki and the onboarding guides to be translated in the future.

### Services going into PROD for the first time
<!-- Required -->

*This can include services going into PROD behind a feature flag that is not turned on.*

Yes the following tools will be deployed:

> - Zoom for video conferencing. Only internal teams will have a Zoom license, but the public will join Zoom   

Reviewer: ```suggestion
- Zoom for video conferencing. Only internal teams will have a Zoom license, but the public will join Zoom
- **Google group:** We'll use this for email-based communication with our open source community.
- **Public wiki:** We'll be using this to share public information about the project, such as onboarding guides, presentations, and meeting notes for public meetings.
```

### Services being integrated in PROD for the first time
<!-- Required -->

*Are there multiple services that are being connected for the first time in PROD?*

Yes, we expect there to be some integrations between the following tools in production:

1. **Chat + Ticket tracking:** Option to receive updates on key tickets in chat
2. **Wiki + Chat:** Option to receive updates on key document changes in chat
3. **Document Sharing + Chat:** Option to show previews of documents shared in chat

> 4. **Sprint Retro + Chat:** Option to share retro takeaways and action items in chat

Reviewer: Not sure if this is still a goal?

5. **Video Conference + Shared Calendar:** Option to add video conference details to events

> 6. **Document Sharing + Wiki:** Option to embed public documents in wiki

Reviewer: Do we have an ADR for document sharing?

Author: We do. Here: https://github.com/HHS/simpler-grants-gov/issues/118

7. **Shared Calendar + Wiki:** Option to embed public calendar events in the wiki

### Data being shared publicly for the first time
<!-- Required -->

*Are there any fields being shared publicly that have never been shared in PROD before?*

> 

Reviewer: ```suggestion
- No, this 30k deliverable does not involve sharing any new production data.
```

### Security considerations
<!-- Required -->

*Does this milestone expose any new attack vectors or expand the attack surface of the product?*

> 

Reviewer: Let's not forget to fill this out! The main security consideration that stands out to me is:

- Adding members of the public to slack, which contains a mix of public and private channels

And the mitigation strategy I'd highlight is:

- We have policy guidelines around where to post sensitive content.
- We've trained internal staff on those policies
- We'll review public channels for sensitive content and remove that content prior to inviting open source contributors.
